nativescript-angular/http-client/ns-http-backend.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from "@angular/core";
+import {
+    HttpRequest, HttpEvent,
+    XhrFactory, HttpResponse,
+    HttpErrorResponse, HttpXhrBackend
+} from "@angular/common/http";
+import { Observable } from "rxjs/Observable";
+import { Observer } from "rxjs/Observer";
+
+import { NSFileSystem } from "../file-system/ns-file-system";
+import { path } from "tns-core-modules/file-system";
+
+@Injectable()
+export class NsHttpBackEnd extends HttpXhrBackend {
+    constructor(xhrFactory: XhrFactory, private nsFileSystem: NSFileSystem) {
+        super(xhrFactory);
+    }
+
+    handle(req: HttpRequest<any>): Observable<HttpEvent<any>> {
+        let result: Observable<HttpEvent<any>>;
+
+        if (isLocalRequest(req.url)) {
+            result = this.handleLocalRequest(req.url);
+        } else {
+            result = super.handle(req);
+        }
+
+        return result;
+    }
+
+    private getAbsolutePath(url: string): string {
+        url = url.replace("~", "").replace("/", "");
+        url = path.join(this.nsFileSystem.currentApp().path, url);
+        return url;
+    }
+
+    private handleLocalRequest(url: string): Observable<HttpEvent<any>> {
```

Comment:
This seems to be the same as `_requestLocalUrl` from `NSHttp` except the additional error handling for failed parsing. Can we move the method to a common module (with the error handling)?